toolkit-test-app/src/main/java/com/esri/arcgisruntime/toolkit/test/bookmark/BookmarkActivity.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2019 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.esri.arcgisruntime.toolkit.test.bookmark
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.ViewModelProviders
+import com.esri.arcgisruntime.mapping.Bookmark
+import com.esri.arcgisruntime.toolkit.bookmark.BookmarkView
+import com.esri.arcgisruntime.toolkit.test.R
+import com.esri.arcgisruntime.toolkit.test.bookmark.map.MapViewModel
+import com.esri.arcgisruntime.toolkit.test.databinding.ActivityBookmarkBinding
+import kotlinx.android.synthetic.main.activity_bookmark.*
+
+class BookmarkActivity : AppCompatActivity(), BookmarkView.OnItemClickListener<Bookmark> {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val binding: ActivityBookmarkBinding =
+            DataBindingUtil.setContentView(this, R.layout.activity_bookmark)
+        val mapViewModel = ViewModelProviders.of(this)[MapViewModel::class.java]
+
+        binding.mapViewModel = mapViewModel
+        binding.lifecycleOwner = this
+
+        mapViewModel.map.addDoneLoadingListener{
+            mapViewModel.updateBookmarks()
+        }
```

Comment:
I guess the MapViewModel itself could listen to the map's load status and then update the bookmarks. I'll leave it to you whether you want to do this here or in MapViewModel.

Reply:
I have moved the addDoneLoadingListener to MapViewModel